### DISCUSSION TITLE
Create a section of an assembly and export it as a DXF file

### DISCUSSION THREAD

**Discussion by Jopie01** [CadQuery/cadquery#1568] [Q&A] [answered]
I'm looking through the code and documentation but cannot find how I can create a section from an assembly and export it as a DXF file.

I have an assembly and want to create a new plane and use that to 'cut' through the assembly. The result then has to be exported to a DXF file (I hope it will be possible to go further and add dimensions to it before exporting).

Can anyone give me some direction how to accomplish this?

**Comment by jmwright** [comment]
@Jopie01 I would wait to hear from one of the other devs because they will likely know of a cleaner way to do this. However, you can round-trip the assembly through the fused STEP exporter to get what you want.

```python
import cadquery as cq

assy = cq.Assembly()
assy.add(cq.Workplane().box(10, 10, 10), color=cq.Color(1.0, 0, 0), name="box1")
assy.add(cq.Workplane().box(5, 5, 5), color=cq.Color(0, 1.0, 0), loc=cq.Location((5.0, 5.0, 5.0), (0, 1, 0), 45), name="box2")

# Round-trip to STEP to completely fuse the assembly
assy.save(
    path=str("temp_assy.step"),
    exportType=cq.exporters.ExportTypes.STEP,
    mode=cq.exporters.assembly.ExportModes.FUSED,
)
fused_assy = cq.importers.importStep("temp_assy.step")

# Do the section
sect = fused_assy.section(3.0)

cq.exporters.export(sect, "temp_dxf.dxf")

show_object(sect)
show_object(assy)
```

![Screenshot from 2024-04-22 12-47-28](https://github.com/CadQuery/cadquery/assets/1015439/4d2db096-ee6a-4657-bd2a-357468c0c339)

**Comment by adam-urbanczyk** [comment, marked as answer]
Not ideal, but maybe like this:
```python
cq.Workplane().add(assy.toCompound()).section()
```

**Comment by Jopie01** [comment]
Thank you both! I'm going with the answer of @adam-urbanczyk because then the parts are not fused into one. I made it even a bit more complex :smile: 

```python
result = cq.Workplane("YZ").workplane(offset=100).add(assy.toCompound()).section()
cq.exporters.export(result, "temp_dxf.dxf")
```

To be honest I thought it would be a cross-section, but this only shows what's on the plane. For now it doesn't matter, but is there a function to do a cross-section?

**Reply by adam-urbanczyk** [reply]
It does generate a cross-section. You can specify the distance from the current workplane like so `.section(0.5)`

**Reply by Jopie01** [reply]
Thanks, that works indeed, make things a bit easier.

But I think we have a miscommunication here. I think I mean clipping, so you remove a part of the 3D model (assembly) to look inside. The cross-section is just whats on the workplane.

**Reply by jmwright** [reply]
@Jopie01 So something more like this?
```python
import cadquery as cq

assy = cq.Assembly()
assy.add(cq.Workplane().circle(10).extrude(5), color=cq.Color(1.0, 0, 0), name="box1")
assy.add(cq.Workplane().circle(5).extrude(5), color=cq.Color(0, 1.0, 0), loc=cq.Location((0.0, 0.0, 5.0), (0, 1, 0), 0), name="box2")

assy_cmpd = assy.toCompound()

section_box= cq.Workplane().move(5, 5).box(10, 10, 20)

show_object(assy_cmpd.cut(section_box.val()))
```
![Screenshot from 2024-04-23 18-14-49](https://github.com/CadQuery/cadquery/assets/1015439/a1213be6-fc57-4b76-819a-1936310605fe)

**Reply by Jopie01** [reply]
Yes, that's what I meant. But instead of cutting out a quarter I want to cut it in "YZ"  plane, eventually with an offset.  I hoped a workplane would be enough but it seems I need also a big box to actually cut.

**Reply by adam-urbanczyk** [reply]
Then use `split`, you can change the workplane (e.g. rotate) like so: https://cadquery.readthedocs.io/en/latest/examples.html#rotated-workplanes . It is good to read the friendly manual every now and then.

NB: CQ DXF export is 2D only, so what you are trying to do (according to the title) will not work in the end.

**Reply by Jopie01** [reply]
Thanks for the reminder but you have to know the functions or their names.

> NB: CQ DXF export is 2D only, so what you are trying to do (according to the title) will not work in the end.

I'm going to use both, the sectioning (my question) for actual exporting and the clipping / splitting for my visualization to see if everything fits.
